### ISSUE TITLE
Dialyzer output

### ISSUE THREAD

**Issue by johnkelly** [elixir-lang/gettext#98]
mix.lock
``` 
"gettext": {:hex, :gettext, "0.11.0"},
```

Relevant mix.exs dialyzer configuration in project function
```
      dialyzer: [
        plt_add_deps: true,
        plt_file: ".local.plt"
      ]
```

On a brand new elixir 1.2.5 phoenix project, I get the following output from the dialyzer.

```
mix dialyzer

gettext.ex:1: The inferred type for the 1st argument of 'MACRO-dgettext'/3 ({_,_}) is not a supertype of #{}, which is expected type for this argument in the callback of the 'Elixir.Gettext.Backend' behaviour
gettext.ex:1: The inferred type for the 1st argument of 'MACRO-dgettext'/4 ({_,_}) is not a supertype of #{}, which is expected type for this argument in the callback of the 'Elixir.Gettext.Backend' behaviour
gettext.ex:1: The inferred type for the 1st argument of 'MACRO-dngettext'/5 ({_,_}) is not a supertype of #{}, which is expected type for this argument in the callback of the 'Elixir.Gettext.Backend' behaviour
gettext.ex:1: The inferred type for the 1st argument of 'MACRO-dngettext'/6 ({_,_}) is not a supertype of #{}, which is expected type for this argument in the callback of the 'Elixir.Gettext.Backend' behaviour
```

**Comment by whatyouhide** [comment]
@josevalim any idea of what's going on here?

**Comment by josevalim** [comment]
Not sure. :(

**Comment by johnkelly** [comment]
When I search phoenix repo for dgettext or dngettext this turns up:

https://github.com/phoenixframework/phoenix/blob/bd4b9984397497a48b99c7b47d8b057c380af71b/installer/templates/new/web/views/error_helpers.ex#L35-L37

Is it because the first argument is an atom?

**Comment by josevalim** [comment]
@johnkelly the typespecs for those seem correct though: https://github.com/elixir-lang/gettext/blob/master/lib/gettext.ex#L501

I am thinking then it may rather be an issue with the macros defined in the Gettext backend?

**Comment by whatyouhide** [comment]
@josevalim is right, dialyzer is complaining about the macros defined by `Gettext` when you call `use Gettext` in your gettext module. As you can see, it mentions `'MACRO-dgettext'/3` and others but `Gettext.*gettext` (which you linked to above) are functions.

I really have no idea what's going on here 😕

**Comment by whatyouhide** [comment]
@ericmj and @fishcakez, I'm gonna go ahead and ping you as well here cause I really don't know where to start. Any ideas?

**Comment by ericmj** [comment]
Sorry, I don't use dialyzer.

**Comment by jeremyjh** [comment]
I put this in my gettext.ex module to silence these warnings.

```Elixir
   @dialyzer [{:nowarn_function, 'MACRO-dgettext': 3},
             {:nowarn_function, 'MACRO-dgettext': 4},
             {:nowarn_function, 'MACRO-dngettext': 5},
             {:nowarn_function, 'MACRO-dngettext': 6},
            ]
```

**Comment by whatyouhide** [comment]
@josevalim do you think it would be reasonable to put that in Gettext backends when we `use Gettext`?

**Comment by josevalim** [comment]
@whatyouhide I think we should rather fix this. @jeremyjh can you provide a simple app that reproduces those warnings so it makes it easier for us to fix this? Thank you!

**Comment by fishcakez** [comment]
This is fixed on Elixir master and the v1.3 branch, so will be included in `v1.4.0` and `v.1.3.2` whenever they happen.
